Rakefile

```diff
@@ -180,6 +215,110 @@ end
+
+  FileUtils.mkdir_p(File.join(@ext_verify_dir, 'bson'))
+
+  # fetch files
+  @files.each do |filename|
+    uri = filename == 'oboe.i' ? URI("#{File.join(oboe_github, 'swig', filename)}") : URI("#{File.join(oboe_github, filename)}")
+    req = Net::HTTP::Get.new(uri)
+    req['Authorization'] = "token #{oboe_token}"
+
+    local_file = File.join(@ext_verify_dir, filename)
+
+    puts "fetching #{filename}"
+    puts "      to #{local_file}"
+
+    res = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') do |http|
+      http.request(req)
+    end
+
+    File.open(local_file, 'wb') { |f| f.puts res.body }
+  end
+end
+
+desc "Fetch oboe files from files.appoptics.com and create swig wrapper"
+task :oboe_files_appoptics_fetch do
+  oboe_version = File.read('ext/oboe_metal/src/VERSION').strip
+  files_appoptics = "https://files.appoptics.com/c-lib/#{oboe_version}"
+
+  FileUtils.mkdir_p(File.join(@ext_dir, 'src', 'bson'))
+
+  # fetch files
+  @files.each do |filename|
+    remote_file = File.join(files_appoptics, 'include', filename)
+    local_file = File.join(@ext_dir, 'src', filename)
+
+    puts "fetching #{remote_file}"
+    puts "      to #{local_file}"
+
+    URI.open(remote_file, 'rb') do |rf|
+      content = rf.read
+      File.open(local_file, 'wb') { |f| f.puts content }
+    end
+  end
+
+  sha_files = ['liboboe-1.0-alpine-x86_64.so.0.0.0.sha256',
+               'liboboe-1.0-x86_64.so.0.0.0.sha256']
+
+  sha_files.each do |filename|
+    remote_file = File.join(files_appoptics, filename)
+    local_file = File.join(@ext_dir, 'lib', filename)
+
+    puts "fetching #{remote_file}"
+    puts "      to #{local_file}"
+
+    URI.open(remote_file, 'rb') do |rf|
+      content = rf.read
+      File.open(local_file, 'wb') { |f| f.puts content }
+    end
+  end
+
+  FileUtils.cd(File.join(@ext_dir, 'src')) do
+    system('swig -c++ -ruby -module oboe_metal -o oboe_swig_wrap.cc oboe.i')
+  end
+end
+
+desc "Verify files"
+task :oboe_verify do
+  oboe_github_fetch
+  @files.each do |filename|
+    puts "Verifying #{filename}"
+
+    sha_1 = Digest::SHA2.file(File.join(@ext_dir, 'src', filename)).hexdigest
+    sha_2 = Digest::SHA2.file(File.join(@ext_verify_dir, filename)).hexdigest
+
+    if sha_1 != sha_2
+      puts "#{filename} from github and files.appoptics.com differ"
+      puts `diff #{File.join(@ext_dir, 'src', filename)} #{File.join(@ext_verify_dir, filename)}`
+      exit 1
+    end
+  end
+end
+
+desc "Build and publish to Rubygems"
+# !!! publishing requires gem >=3.0.5 !!!
+# Don't run with Ruby versions < 2.7 they have gem < 3.0.5
+task :build_and_publish_gem do
+  gemspec_file = 'appoptics_apm.gemspec'
+  gemspec = Gem::Specification.load(gemspec_file)
+  gem_file = gemspec.full_name + '.gem'
+
+  exit 1 unless system('gem', 'build', gemspec_file)
+
+  if ENV['GEM_HOST_API_KEY']
+    exit 1 unless system('gem', 'push', gem_file)
+  end
```

Comment:
If the api key env var is not set, does this task just exit with OK status even though the gem has not actually been published?

Reply:
yes! I thought it would make it a bit harder to accidentally push a gem.